sql/parser/eval.go

```diff
@@ -625,6 +625,9 @@ func EvalExpr(expr Expr) (Datum, error) {
 		return DString(t), nil
 
 	case IntVal:
+		if t < 0 {
+			return DNull, fmt.Errorf("integer value out of range: %d", uint64(t))
```

Comment:
Well, `IntVal` is really storing an unsigned integer even though its underlying type is `int64`. I don't want to display the negative value. I think I can just use `%s` and rely on `IntVal.String`, though.